### DISCUSSION TITLE
What's the easiest way to set metadata in a client layout?

### DISCUSSION THREAD

**Discussion by vedantroy** [vercel/next.js#50883] [Help] [answered]
### Summary

I need my root layout to be a client component because I am initializing things like mixpanel/logrocket inside of it.
What is the easiest way to set metadata in this case?

**Comment by icyJoseph** [comment, marked as answer]
Refactor so that root layout is still a server component.

You can do like:

```tsx
import "./globals.css";
import { Inter } from "next/font/google";
import { SetupClientSideStuff } from "./SetupClientSideStuff";

const inter = Inter({ subsets: ["latin"] });

/* setup meta data stuff here */

export default function RootLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return (
    <html lang="en">
      <SetupClientSideStuff />
      <body className={inter.className}>{children}</body>
    </html>
  );
}
```

And that `SetupClientSideStuff` does all of the things you want. It might be that you need to target the `html` or `body` tags though, and that'll have its own small challenges, but then you'd just need to collect all of that into a Client Component that is imported into the root layout file. That allows you to keep the layout as a server component.

**Reply by vedantroy** [reply]
Got it working, with a client-only component for my metrics:
```
export default function () {
  const logRocketID = process.env.NEXT_PUBLIC_LOGROCKET_ID;
  useEffect(() => {
    if (logRocketID) {
      console.log("LogRocket ID found -- initializing")
      LogRocket.init(logRocketID);
    } else {
      console.log("LogRocket ID not found")
    }
  }, [])
  return <></>
}
```

It is a bit odd to return an empty react component though, although I think it works.

**Reply by icyJoseph** [reply]
Nice! Do you ever experience Fast Refresh unexpected behavior? IIRC, anonymous exports, like `export default () => {}` used to cause state loss.

- https://nextjs.org/docs/architecture/fast-refresh#limitations
